README.md

```diff
@@ -235,12 +251,36 @@ fast-jwt supports caching of verified tokens.
 
 The cache layer, powered by [mnemonist](https://www.npmjs.com/package/mnemonist), is a LRU cache which dimension is controlled by the user, as described in the options list.
 
-When caching is enabled, verified tokens are always stored in cache. If the verification fails once, the error is cached as well and the operation is not retried.
+When caching is enabled, verified tokens are always stored in cache. If the verification fails once, the error is cached as well for the time set by `errorCacheTTL` and the operation is not retried.
 
 For verified tokens, caching considers the time sensitive claims of the token (`iat`, `nbf` and `exp`) and make sure the verification is retried after a token becomes valid or after a token becomes expired.
 
 Performances improvements varies by uses cases and by the type of the operation performed and the algorithm used.
 
+## Token Error Codes
+
+This is the lisf of the error codes exported by `TOKEN_ERROR_CODES`:
+
+| Code                              | Description                                                 |
+| --------------------------------- | ----------------------------------------------------------- |
+| `FAST_JWT_INVALID_TYPE`           | Invalid token type                                          |
+| `FAST_JWT_INVALID_OPTION`         | The option object is not valid                              |
+| `FAST_JWT_INVALID_ALGORITHM`      | The token algorithm is invalid                              |
+| `FAST_JWT_INVALID_CLAIM_TYPE`     | The claim value is not supported                            |
+| `FAST_JWT_INVALID_CLAIM_VALUE`    | The claim type is not a positive integer or an number array |
+| `FAST_JWT_INVALID_KEY`            | The key is not a string or a buffer or is unsupported       |
+| `FAST_JWT_INVALID_SIGNATURE`      | The token signature is invalid                              |
+| `FAST_JWT_INVALID_PAYLOAD`        | The payload to be decoded must be an object                 |
+| `FAST_JWT_MALFORMED`              | The token is malformed                                      |
+| `FAST_JWT_INACTIVE`               | The token is not valid yet                                  |
+| `FAST_JWT_EXPIRED`                | The token is expired                                        |
+| `FAST_JWT_MISSING_KEY`            | The key option is missing                                   |
+| `FAST_JWT_KEY_FETCHING_ERROR`     | Could not retrieve the key                                  |
+| `FAST_JWT_SIGN_ERROR`             | Cannot create the signature                                 |
+| `FAST_JWT_VERIFY_ERROR`           | Cannot verify the signature                                 |
+| `FAST_JWT_MISSING_REQUIRED_CLAIM` | A required claim is missing                                 |
+| `FAST_JWT_MISSING_SIGNATURE`      | The token signature is missing                              |
```

Comment:
why don't we keep this exclusively in the code and link to the code here? so that we don't have to keep this up to date

Reply:
did you include comments next to the errors as were in the table? so they also serve as documentation

Reply:
yes that's smart